### PR TITLE
[ADI] Add votes count by type

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Controller/EnMarche/SearchController.php [hunk 1]
```diff
@@ -23,7 +24,7 @@ class SearchController extends Controller
      * @Route("/evenements/categorie/{slug}", name="app_search_events_by_category")
      * @Method("GET")
      */
-    public function searchEventsAction(Request $request, string $slug = null)
+    public function searchEventsAction(IdeaRepository $rep, Request $request, string $slug = null)
```
Comment (ottaviano) [inline src/Controller/EnMarche/SearchController.php]: can you remove it pls

Comment (thomasbisignani) [inline src/Controller/EnMarche/SearchController.php]: This repository is not used

File: src/Repository/IdeaRepository.php [hunk 2]
```diff
@@ -49,4 +50,46 @@ public function countThreadComments(Idea $idea): int
...
+        ;
+
+        return array_reduce($votes, function ($result, $vote) {
+            $result[$vote['type']] = $vote['count'];
+
+            return $result;
+        }, []);
+    }
+
+    public function countAdherentVotesForIdea(Idea $idea, Adherent $adherent): array
```
Comment (ottaviano) [inline src/Repository/IdeaRepository.php]: what about `getAdherentVoteForIdea` and return Vote objects?

File: src/Repository/IdeaRepository.php [hunk 3]
```diff
@@ -49,4 +50,46 @@ public function countThreadComments(Idea $idea): int
...
+    {
+        $votes = $this
+            ->createQueryBuilder('idea')
+            ->select('DISTINCT vote.type')
```
Comment (ottaviano) [inline src/Repository/IdeaRepository.php]: why DISTINCT? you already have a constraint on idea, author and type, no?

File: src/Controller/EnMarche/SearchController.php (hunk 1 above)
Comment (illusionOfParadise) [reply src/Controller/EnMarche/SearchController.php]: Done.

Comment (illusionOfParadise) [reply src/Controller/EnMarche/SearchController.php]: Done.

File: src/Repository/IdeaRepository.php [hunk 4]
```diff
@@ -49,4 +50,46 @@ public function countThreadComments(Idea $idea): int
...
+            ->select('vote.type, COUNT(idea) as count')
+            ->innerJoin('idea.votes', 'vote')
+            ->where('idea = :idea')
+            ->setParameter('idea', $idea)
+            ->groupBy('vote.type')
+            ->getQuery()
+            ->getArrayResult()
+        ;
+
+        return array_reduce($votes, function ($result, $vote) {
```
[the remark this answers is not in the document]
Comment (illusionOfParadise) [reply src/Repository/IdeaRepository.php]: Done.

File: src/Repository/IdeaRepository.php (hunk 2 above)
Comment (illusionOfParadise) [reply src/Repository/IdeaRepository.php]: Ok for renaming, but It is not in the interest to return Vote objects.

File: src/Serializer/IdeasWorkshopSerializerNormalizer.php [hunk 5]
```diff
@@ -25,6 +31,17 @@ public function normalize($object, $format = null, array $context = [])
         if (\in_array('idea_list_read', $context['groups'])) {
             $data['contributors_count'] = $this->ideaRepository->countIdeaContributors($object);
             $data['comments_count'] = $this->ideaRepository->countThreadComments($object);
+
+            $total = $data['votes_count'];
+            $data['votes_count'] = ['total' => $total];
+            $votes = $this->ideaRepository->countVotesByType($object);
+            foreach (VoteTypeEnum::toArray() as $type) {
```
Comment (ottaviano) [inline src/Serializer/IdeasWorkshopSerializerNormalizer.php]: you don't need to do this anymore, as your repo return 0 value for others types

**Pull request merged**